figures/S7.supplemental_figure7/scripts/S7.r

```diff
@@ -0,0 +1,288 @@
+    "figures"
+)
+if (!dir.exists(figure_dir)) {
+    dir.create(figure_dir)
+}
+
+# sort the index file by labeled_data_index
+index_file_PBMC <- index_file_PBMC[order(index_file_PBMC$labeled_data_index),]
+head(index_file_PBMC,2)
+index_file_SHSY5Y <- index_file_SHSY5Y[order(index_file_SHSY5Y$labeled_data_index),]
+head(index_file_SHSY5Y,2)
+
+
+# set path to the PBMC metadata file
+metadata_file_path_PBMC <-file.path(
+    "..",
+    "..",
+    "..",
+    "data",
+    "PBMC_preprocessed_sc_norm.parquet"
+)
+# load the metadata file via arrow
+# metadata_file <- arrow::read_parquet(metadata_file_path)
+
+# load via arrow with certain columns
+metadata_file_PBMC <- arrow::read_parquet(
+    metadata_file_path_PBMC,
+    col_select = c(
+        'Metadata_cell_type',
+        'Metadata_Well'	,
+        'Metadata_number_of_singlecells',
+    )
+)
+head(metadata_file_PBMC)
+
+# SHSY5Y
+# set path to the SHSY5Y metadata file
+metadata_file_path_SHSY5Y <-file.path(
+    "..",
+    "..",
+    "..",
+    "data",
+    "SHSY5Y_preprocessed_sc_norm.parquet"
+)
+# load the metadata file via arrow
+metadata_file_SHSY5Y <- arrow::read_parquet(
+    metadata_file_path_SHSY5Y,
+    col_select = c(
+        'Metadata_cell_type',
+        'Metadata_Well'	,
+        'Metadata_number_of_singlecells',
+    )
+)
+head(metadata_file_SHSY5Y)
+
+
+
+
+
+## PBMC
+
+# add column to metadata file with the labeled_data_index
+metadata_file_PBMC$label <- index_file_PBMC$label
+# replace test, train, or validation with "pool" in label column
+metadata_file_PBMC$label <- gsub(
+    pattern = "test",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+metadata_file_PBMC$label <- gsub(
+    pattern = "train",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+metadata_file_PBMC$label <- gsub(
+    pattern = "val",
+    replacement = "pool",
+    x = metadata_file_PBMC$label
+)
+# aggreagate the metadata via Metadata_Well
+metadata_file_PBMC <- metadata_file_PBMC %>%
+    group_by(Metadata_Well, Metadata_cell_type, label) %>%
+    summarize(
+        n = n()
+    )
+head(metadata_file_PBMC)
+
+# SHSY5Y
+# add column to metadata file with the labeled_data_index
+metadata_file_SHSY5Y$label <- index_file_SHSY5Y$label
+# replace test, train, or validation with "pool" in label column
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "test",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "train",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+metadata_file_SHSY5Y$label <- gsub(
+    pattern = "val",
+    replacement = "pool",
+    x = metadata_file_SHSY5Y$label
+)
+# aggreagate the metadata via Metadata_Well
+metadata_file_SHSY5Y <- metadata_file_SHSY5Y %>%
+    group_by(Metadata_Well, Metadata_cell_type, label) %>%
+    summarize(
+        n = n()
+    )
+head(metadata_file_SHSY5Y)
+
+
+# combine the SHSY5Y and PBMC metadata files
+combined_metadata <- rbind(
```

Comment:
By the time `metadata_file_PBMC` and `metadata_file_SHSY5Y` arrive here, what datatype are they? Would it make sense to use something besides `rbind` to combine them (for instance, [combine via `c()`](https://arrow.apache.org/docs/r/articles/dataset.html#work-with-multiple-data-sources))?

Reply:
They are data frames so rbind makes the most sense to me